cpp/src/lance/io/reader.cc

```diff
@@ -289,6 +289,7 @@ ::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> FileReader::ReadBatch(
 
 ::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> FileReader::ReadBatch(
     const lance::format::Schema& schema, int32_t batch_id, const ArrayReadParams& params) const {
+  assert(!schema.fields().empty());
```

Comment:
Does this raise exception if false or just segfault?

Reply:
good point. it does segfault in debug mode. 

I considered that this should be a condition that never happens, thus as a bug.